sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java

```diff
@@ -198,6 +201,33 @@ public void testSelectsFromExistingTable() throws Exception {
     assertThat(resultRows, containsInAnyOrder(row(1L, "aaa"), row(2L, "bbb")));
   }
 
+  @Test
+  public void testTimestamp() throws Exception {
+    TestTableProvider tableProvider = new TestTableProvider();
+    Connection connection = JdbcDriver.connect(tableProvider);
+
+    // A table with one TIMESTAMP column
+    Schema schema = Schema.builder().addDateTimeField("ts").build();
+    connection.createStatement().executeUpdate("CREATE TABLE test (ts TIMESTAMP) TYPE 'test'");
+
+    ReadableInstant july1 =
+        ISODateTimeFormat.dateTimeParser().parseDateTime("2018-07-01T01:02:03Z");
+    tableProvider.addRows("test", Row.withSchema(schema).addValue(july1).build());
+
+    ResultSet selectResult =
+        connection.createStatement().executeQuery(String.format("SELECT ts FROM test"));
+    selectResult.next();
+    Timestamp ts = selectResult.getTimestamp(1);
+
+    assertThat(
```

Comment:
Regarding my comment: this test fails on my laptop, off by the size of the timezone. So I think there's some locale-dependent path here.